azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -86,4 +97,26 @@ protected void configure() {
   LocalStorage createLocalStorage(StorageConfig config) {
     return new LocalStorage(new File(config.getBaseDirectoryPath()));
   }
+
+  // todo kunkun-tang: the below method should moved out to azkaban-db module eventually.
+  // Today azkaban-db can not rely on Props, so we can not do it.
+  private static AzkabanDataSource getDataSource(Props props) {
```

Comment:
suggestion: remove static and make this a provider like the method just above this `createLocalStorage` The idea is to initialize only the minimal set of classes like config classes etc and leave the rest to Guice.